internal/database/gitserver_repos.go

```diff
@@ -704,3 +711,22 @@ func sanitizeToUTF8(s string) string {
 	// Sanitize to a valid UTF-8 string and return it.
 	return strings.ToValidUTF8(t, "")
 }
+
+func (s *gitserverRepoStore) SetCloningProgress(ctx context.Context, repoName api.RepoName, progressLine string) error {
+	res, err := s.ExecResult(ctx, sqlf.Sprintf(`
+UPDATE gitserver_repos
+SET
+	cloning_progress = %s,
+	updated_at = NOW()
+WHERE repo_id = (SELECT id FROM repo WHERE name = %s)
+`, progressLine, repoName))
+	if err != nil {
+		return errors.Wrap(err, "setting cloning progress")
+	}
+	if nrows, err := res.RowsAffected(); err != nil {
+		return errors.Wrap(err, "getting rows affected")
+	} else if nrows != 1 {
+		return errors.New("repo not found")
```

Comment:
I would suggest adding more context on this error. Just seeing `repo not found` could raise some false alarms.

```suggestion
		return errors.Newf("failed to set cloning progress, repo %q not found", repoName)
```